presto-common/src/main/java/com/facebook/presto/common/block/BlockUtil.java

```diff
@@ -106,6 +108,17 @@ else if (newSize > MAX_ARRAY_SIZE) {
         return (int) newSize;
     }
 
+    static int calculateBlockResetSize(int currentSize, int currentPositionCount, int expectedPositionCount)
```

Comment:
Having currentSize currentPositionCount parameters is confusing. In the original calculateBlockResetSize method currentSize is current-position-count, but here it is not. Maybe renaming the method to calculateNestedBlockResetSize would help. Parameter still need to be renamed, but I don't have any specific suggestions just yet.

Reply:
Thanks @mbasmanova for the suggestions. This method is intended to be used in calculating either the nested block position count, or VariableWidthBlockBuilder's slice size. So I renamed it `calculateNestedStructureResetSize` and renamed the parameters as well:
```
static int calculateNestedStructureResetSize(int currentNestedStructureSize, int currentNestedStructurePositionCount, int expectedPositionCount)
    {
        long newSize = max(
                (long) ceil(currentNestedStructureSize * BLOCK_RESET_SKEW),
                currentNestedStructurePositionCount == 0 ? currentNestedStructureSize : (long) currentNestedStructureSize * expectedPositionCount / currentNestedStructurePositionCount);
        if (newSize > MAX_ARRAY_SIZE) {
            return MAX_ARRAY_SIZE;
        }
        return toIntExact(newSize);
    }
```
Do you think this is easier to understand?

btw.There were two existing overload methods
```
static int calculateBlockResetSize(int currentSize)
    {
        long newSize = (long) ceil(currentSize * BLOCK_RESET_SKEW);

        // verify new size is within reasonable bounds
        if (newSize < DEFAULT_CAPACITY) {
            newSize = DEFAULT_CAPACITY;
        }
        else if (newSize > MAX_ARRAY_SIZE) {
            newSize = MAX_ARRAY_SIZE;
        }
        return (int) newSize;
    }

    static int calculateBlockResetBytes(int currentBytes)
    {
        long newBytes = (long) ceil(currentBytes * BLOCK_RESET_SKEW);
        if (newBytes > MAX_ARRAY_SIZE) {
            return MAX_ARRAY_SIZE;
        }
        return (int) newBytes;
    }
```
But they're doing the same thing and I actually think the calculateBlockResetBytes(currentBytes) should be removed.

Reply:
> But they're doing the same thing and I actually think the calculateBlockResetBytes(currentBytes) should be removed.

@yingsu00 I agree.

Reply:
> > But they're doing the same thing and I actually think the calculateBlockResetBytes(currentBytes) should be removed.
> 
> @yingsu00 I agree.

@mbasmanova ok, I added a new commit "Remove calculateBlockResetBytes in BlockUtil" to this PR that removes it.
How do you like the new `calculateNestedStructureResetSize(int currentNestedStructureSize, int currentNestedStructurePositionCount, int expectedPositionCount)
` naming?